src/EditorFeatures/Core/Implementation/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs

```diff
@@ -480,14 +481,34 @@ public void ExecuteCommand(ReturnKeyCommandArgs args, Action nextHandler)
             // Check to see if the current line starts with exterior trivia. If so, we'll take over.
             // If not, let the nextHandler run.
 
-            var caretPosition = args.TextView.GetCaretPoint(args.SubjectBuffer) ?? -1;
-            if (caretPosition < 0)
+            int originalPosition;
+
+            // The original position should be a position that is consistent with the syntax tree, even
+            // after Enter is pressed. Thus, we use the start of the first selection if there is one.
+            // Otherwise, getting the tokens to the right or the left might return unexpected results.
+
+            if (args.TextView.Selection.SelectedSpans.Count > 0)
+            {
+                var selectedSpan = args.TextView.Selection
+                    .GetSnapshotSpansOnBuffer(args.SubjectBuffer)
+                    .FirstOrNullable();
+
+                originalPosition = selectedSpan != null
+                    ? args.TextView.BufferGraph.MapUpOrDownToBuffer(selectedSpan.Value.Start, args.SubjectBuffer) ?? -1
```

Comment:
`GetSnapshotSpansOnBuffer` already did the mapping. Why is the mapping happening again?